test/trento/databases/database_test.exs

```diff
@@ -247,17 +257,23 @@ defmodule Trento.Databases.DatabaseTest do
     end
 
     test "should not add a database instance if the database instance was already registered" do
-      database_registered_event = build(:database_registered_event)
+      tenants = build_list(1, :tenant)
+
+      %{database_id: database_id} = database_registered_event = build(:database_registered_event)
 
       database_instance_registered_event =
         build(
           :database_instance_registered_event,
           database_id: database_registered_event.database_id
         )
 
+      tenants_updated_event =
```

Comment:
Do you need to provide an external `tenants`?
The factory should create it for you.
The same for every other test. You create the tenants before to pass it to a factory that create them by default

Reply:
yeah it's true I did like that to mantain the compatibility with the old way to to tests and not change everything and assign to a variable, I will change some of that